### ISSUE TITLE
Installing through the "install from zip" option

### ISSUE THREAD

**Issue by pitsi** [Colossal1/plugin.video.SportsDevil#7]
I have been installing sportsdevil by downloading the latest zip file and then via kodi's install from zip option since... forever. Why? Because of these reasons
- sportsdevil was not always in a repo, so this was the only way of installing it
- as of sportsdevil 20-03-2017, the addon starts some proxy service (line 22 on addon.xml) which either made my kodi hang on exit or report that sportsdevil was not installed properly. Sadly, the relevant log no longer exists, because pastebin's expiration date has passed.

So, although I have no problem installing the colossus repo and then sportsdevil from it, I would like to have control over that single line and remove it before installation/upgrade. So, can I use github's "download zip" function (green "clone or download" button > "download zip" option) to download this part of the repo as a zip and install it as usual? Or is itnot recommended or dangerous or something else?

**Comment by Colossal1** [comment]
You can always disable autoupdate for the addon or remove the repo. And of course you can still download zip files and install from there. If the proxy service causes problems, simply comment out that line 22.

**Comment by pitsi** [comment]
One at a time...

Disabling autoupdate for sportsdevil is pointless, because my problem is not just manual updating. It's downloading, then editing the file, then updating. And in that procedure kodi won't let me do the editing part as will update it the second it downloads the zip.

Removing the repo is also pointless, for the obvious reason.

And then comes my github... habit. I have been using github long before I made an account here, which in fact was made for making issue reporting only, and I currently monitor ~20 projects here. One important thing I have learned is to NOT use the "download zip" function or the "git-clone" command on my terminal unless I am REALLY sure I want the under-development version of the project, because it is almost certain it will contain bugs, which I won't be able to fix on my own.
I only download stuff that contains release-ready code and is listed under the "releases" tab on the project's github page, e.g. youtube-dl's release page

https://github.com/rg3/youtube-dl/releases

So, since plugin.video.sportsdevil does not have anything under the releases tab, I hesitate to use the download zip function, because if I come across a bug, I won't be able to fix it. So, if you can, please add every release-ready version of sportsdevil there as well.

**Comment by Gujal00** [comment]
@pitsi You are not meant to use the download zip function (it is useless) or the git-clone.
If you delete the repo, You are required to manually get the zip file stored in this folder
https://github.com/Colossal1/repository.colossus.common/tree/master/plugin.video.SportsDevil

This zip file you can install from from zip within kodi and once installed you can go and comment the line 22

**Comment by pitsi** [comment]
Thanks for showing me the folder that contains the zip itself. I will use it next time instead of doing what I said above. I will also try installig the repo and then sportsdevil from it, and let's hope it won't hang this time. 

Doing all that in a pc, where you have the ability to kill kodi if it does not respond after installing sportsdevil and removing the bad line in a proper text editor like geany, is no big deal. 
However, doing the same procedure in a device like rpi with an os like openelec installed, where you have to kill kodi and stop kodi.service from relaunching through systemd, and then edit the file with nano, that is a pain in the ***.

**Comment by Gujal00** [comment]
I use Libreelec on S802 myself, and is the easiest thing to do. Just ssh to the box and edit file and reboot. No need to kill any hanging kodi and modifying systemd etc. Why the drama? ;-)

**Comment by pitsi** [comment]
True, it is kinda drama. But that single line, back when sportsdevil 2017-03-20 was released, made 2 of my kodi 17 installations hang (one on my linux pc, one on the rpi zero with openelec).
 
For the one on the pc I became mad, because I could not roll back to 16.x (my distro only had 17.x in the repos) where sportsdevil installed and worked flawlessly. For the one on the rpi, I reinstalled the previous openelec version with kodi 16, because I really needed one working sportsdevil instance. I was "forced to update" it to openelec with kodi 17.x when that subtitles security flaw appeared and the kodi team patched nothing more than 17.x.

And back then it was that I realised tha MANY addons can run as services, hogging cpu and memory for no real reason (no, syncing covenant with trakt every 5 minutes is not a serious reason to have covenant running as a service in the background). 

Now, I have installed the repo and sportsdevil from there and, so far, everything is fine.
